core/src/kura.rs

```diff
@@ -771,37 +771,28 @@ impl BlockStore<Locked> {
 
 type Result<T, E = Error> = std::result::Result<T, E>;
 /// Error variants for persistent storage logic
-#[derive(thiserror::Error, Debug)]
+#[derive(thiserror::Error, Debug, displaydoc::Display)]
 pub enum Error {
-    /// Generic IO error
-    #[error("Failed reading/writing {_1:?} from disk")]
+    /// Failed reading/writing {1:?} from disk
     IO(#[source] std::io::Error, PathBuf),
-    /// Failed to create the directory
-    #[error("Failed to create the directory {_1:?}")]
+    /// Failed to create the directory {1:?}
     MkDir(std::io::Error, PathBuf),
```

Comment:
This will probably omit underlying error in display impl. Can you, please, refactor that? Something like: 

```rust
MkDr {
    source: std::io::Error,
    path: PathBuf,
}
```

Reply:
`thiserror` should be smart enough to handle `source` field, but better to check the docs